habitat/datasets/pointnav/pointnav_dataset.py

```diff
@@ -99,6 +99,9 @@ def __init__(self, config: Optional[Config] = None) -> None:
             with gzip.open(scene_filename, "rt") as f:
                 self.from_json(f.read(), scenes_dir=config.SCENES_DIR)
 
+        if config.NUM_EPISODE_SAMPLE >= 0:
```

Comment:
This can probably be optimized. Instead of loading all of the episodes and then sampling one from them, is there other way round where we only load the required episode?

Reply:
Asked Oleksandr before, and will discuss again. My current understanding is that the single-episode option will likely be used with a small dataset to begin with, e.g. habitat_test_scenes, and we don't care that much about the cost of loading such a small dataset in its entirety.

Reply:
Cool. From developer's perspective, a lot of times I will be testing it, so I need a faster alternative, but then again it depends on which direction the project is going in.

Reply:
@apsdehal current dataset format is gzip json that can't be loaded partially. For large datasets we have separate file per scene. 1-10k (usual size for val split) episodes are really fast to load as file size is several kb.

Reply:
@JasonJiazhiZhang let's update EQADataset as well accordingly.

Reply:
@JasonJiazhiZhang this checks not needed anymore.